### PR TITLE
REF: simplify Grouping.__init__

### PR DESCRIPTION
**Pull request by jbrockmendel**
just de-nesting conditions in `__init__`

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/groupby/grouper.py [hunk 1]
```diff
@@ -500,34 +500,33 @@ def __init__(
...
-                    )
-                    self.grouper = None  # Try for sanity
-                    raise AssertionError(errmsg)
+                grper = pprint_thing(self.grouper)
+                errmsg = (
+                    "Grouper result violates len(labels) == "
+                    f"len(data)\nresult: {grper}"
+                )
+                self.grouper = None  # Try for sanity
+                raise AssertionError(errmsg)
```
Comment (jreback) [inline pandas/core/groupby/grouper.py:529]: OT is this ever hit in tests? we should not be raising an AssertionError

**Review by jreback — APPROVED** [review]
small comment for followon

**Pull request merged**